### PR TITLE
fix: codesplit cli code to allow mud.config.ts to be imported in the browser

### PR DESCRIPTION
**Pull request by alvrs**
We need to be able to import `mud.config.ts` in browser code to infer types for various client libraries (eg `store-cache`).
This PR fixes a couple issues that prevented this:
* Importing `mudConfig` also imported code intended for node environments, which caused `vite` in consumer projects to error (example: 
image)
* `global` (used for mud plugins extending the config) was not defined in browser contexts: 
image

### CODE REVIEW

**Comment by alvrs** [comment]
Unclear why tests are failing, all are passing locally and there were no changes to the store package that would influence the forge tests. Trying `foundryup` to see if it's caused by a new foundry version.

**Comment by holic** [comment]
maybe time for? https://github.com/latticexyz/mud/issues/746

**Comment by holic** [comment]
> Unclear why tests are failing, all are passing locally and there were no changes to the store package that would influence the forge tests. Trying `foundryup` to see if it's caused by a new foundry version.

Yeah, it might be this: https://github.com/foundry-rs/foundry/pull/4920

> The intended behavior is that it only works for the next immediate call. This new behavior only allows to:
>
> - Call the cheatcode AND fill the expected emit right after calling the cheatcode. Declaring the cheatcode N times and then filling N events will fail.
> - match events after the next CALL is triggered. Events not triggered by a call will be ignored. Events nested in the triggered CALL subtree will be matched. If the events defined are not matched, the execution will revert.

It's a little unclear by the above "nested in the CALL subtree" description if this should succeed or fail in our case, but I noticed that `StoreCore.pushToField` is doing more calls internally before emitting, so maybe it's not matching on the new criteria for `expectEmit`?

This feels like a bummer constraint if so, and wonder if there are ways around this or alternative vm helpers.

**Comment by holic** [comment]
Another option is to use Vulcan test utils, which has some nicer ergonomics: https://github.com/nomoixyz/vulcan

```solidity
mc.doSomethingElse();

// Check event emissions
expect(address(mc).lastCall()).toHaveEmitted(
    "SomeEvent(address,bytes32,uint256)",
    [address(1).topic(), any()], // Event topics
    abi.encode(123) // Event data
);
```

(I also have an open issue and they have a pending PR for better gas reporting)

**Comment by holic** [comment]
opened an issue for this since its unrelated to this PR: https://github.com/latticexyz/mud/issues/814

File: .github/workflows/templates.yml [hunk 1]
```diff
@@ -35,5 +35,5 @@ jobs:
         uses: ./.github/actions/require-empty-diff
 
       - name: Run tests
-        working-directory: ./examples/${{ matrix.example }}
+        working-directory: ./templates/${{ matrix.template }}
```
Comment (holic) [inline .github/workflows/templates.yml:38]: nice catch

File: packages/config/package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
   "type": "module",
   "exports": {
     ".": "./dist/library/index.js",
-    "./register": "./dist/register/index.js"
+    "./register": "./dist/register/index.js",
+    "./cli": "./dist/cli/index.js"
```
Comment (holic) [inline packages/config/package.json:15]: should this be `/node`?

**Review by holic — APPROVED** [review]
Looks good! Only question is if we should call this export `node` rather than `cli`, if node is the intended environment. `cli` sounds too use case specific and very internally oriented, and if that's what we're aiming for, maybe we name it `cli-for-internal-use-only` haha

Separately wondering if we should just bite the bullet on #746 (maybe not before the hackathon but quickly following?)

**Review by holic — APPROVED** [review]
Looks good! Only question is if we should call this export `node` rather than `cli`, if node is the intended environment. `cli` sounds too use case specific and very internally oriented, and if that's what we're aiming for, maybe we name it `cli-for-internal-use-only` haha

Separately wondering if we should just bite the bullet on #746 (maybe not before the hackathon but quickly following?)

**Review by holic — DISMISSED** [review]
Looks good! Only question is if we should call this export `node` rather than `cli`, if node is the intended environment. `cli` sounds too use case specific and very internally oriented, and if that's what we're aiming for, maybe we name it `cli-for-internal-use-only` haha

Separately wondering if we should just bite the bullet on #746 (maybe not before the hackathon but quickly following?)

File: packages/config/package.json (hunk 2 above)
Comment (alvrs) [reply packages/config/package.json]: yeah I was torn between the two, but now agree `node` fits better - changed it for both config and world exports

**Comment by alvrs** [comment]
> Separately wondering if we should just bite the bullet on https://github.com/latticexyz/mud/issues/746 (maybe not before the hackathon but quickly following?)

Unfortunately I think it's too close to the hackathon now to refactor this, but agree with shortly after!

**Review by dk1a — APPROVED** [review]
I like `node`

**Review by holic — APPROVED** [review]

**Pull request merged**